src/rng/backends/common/hiplike_helper.hpp

```diff
@@ -0,0 +1,208 @@
+ * so.
+ ******************************************************************************/
+#ifndef _MKL_RNG_HIPLIKE_HELPER_HPP_
+#define _MKL_RNG_HIPLIKE_HELPER_HPP_
+
+namespace oneapi {
+namespace mkl {
+namespace rng {
+
+// Static template functions oneapi::mkl::rng::curand::range_transform_fp for
+// Buffer and USM APIs
+//
+// cuRAND has no built-in functionality to specify a custom range for sampling
+// random numbers; `curandGenerateUniform' generates uniform random numbers on
+// [0, 1). This function is used to convert to range [a, b).
+//
+// Supported types:
+//      float
+//      double
+//
+// Input arguments:
+//      queue - the queue to submit the kernel to
+//      a     - range lower bound (inclusive)
+//      b     - range upper bound (exclusive)
+//      r     - buffer to store transformed random numbers
+template <typename T>
+static inline void range_transform_fp(cl::sycl::queue& queue, T a, T b, std::int64_t n,
+                                      cl::sycl::buffer<T, 1>& r) {
+    queue.submit([&](cl::sycl::handler& cgh) {
+        auto acc = r.template get_access<cl::sycl::access::mode::read_write>(cgh);
+        cgh.parallel_for(cl::sycl::range<1>(n),
+                         [=](cl::sycl::id<1> id) { acc[id[0]] = acc[id[0]] * (b - a) + a; });
+    });
+}
+template <typename T>
+static inline cl::sycl::event range_transform_fp(cl::sycl::queue& queue, T a, T b, std::int64_t n,
+                                                 T* r) {
+    return queue.submit([&](cl::sycl::handler& cgh) {
+        cgh.parallel_for(cl::sycl::range<1>(n),
+                         [=](cl::sycl::id<1> id) { r[id[0]] = r[id[0]] * (b - a) + a; });
+    });
+}
+template <typename T>
+static inline void range_transform_fp_accurate(cl::sycl::queue& queue, T a, T b, std::int64_t n,
+                                               cl::sycl::buffer<T, 1>& r) {
+    queue.submit([&](cl::sycl::handler& cgh) {
+        auto acc = r.template get_access<cl::sycl::access::mode::read_write>(cgh);
+        cgh.parallel_for(cl::sycl::range<1>(n), [=](cl::sycl::id<1> id) {
+            acc[id[0]] = acc[id[0]] * (b - a) + a;
+            if (acc[id[0]] < a) {
+                acc[id[0]] = a;
+            }
+            else if (acc[id[0]] > b) {
+                acc[id[0]] = b;
+            }
+        });
+    });
+}
+template <typename T>
+static inline cl::sycl::event range_transform_fp_accurate(cl::sycl::queue& queue, T a, T b,
+                                                          std::int64_t n, T* r) {
+    return queue.submit([&](cl::sycl::handler& cgh) {
+        cgh.parallel_for(cl::sycl::range<1>(n), [=](cl::sycl::id<1> id) {
+            r[id[0]] = r[id[0]] * (b - a) + a;
+            if (r[id[0]] < a) {
+                r[id[0]] = a;
+            }
+            else if (r[id[0]] > b) {
+                r[id[0]] = b;
+            }
+        });
+    });
+}
+
+// Static template functions oneapi::mkl::rng::curand::range_transform_int for
```

Comment:
curand->rocrand in this file, or should it be common for both curand and rocrand, then probably it makes sense to put it into common/ includes on src/rng level

Reply:
Yes, this file is common for both the rocRAND and the cuRAND backend. I wonder, wouldn't it make more sense to put into /include/oneapi/mkl/rng/detail?

Reply:
Yes, I think it is also fine to do, as it may be used for any other implementations, which doesn't have such scaling